src/category_theory/simple.lean

```diff
@@ -0,0 +1,46 @@
+/-
+Copyright (c) 2020 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+import category_theory.limits.shapes.zero
+import category_theory.limits.shapes.kernels
+
+open category_theory.limits
+
+namespace category_theory
+
+universes v u
+variables {C : Type u} [category.{v} C]
+variables [has_zero_morphisms.{v} C]
+
+/-- An object is simple if monomorphisms into it are (exclusively) either isomorphisms or zero. -/
+class simple (X : C) :=
+(mono_is_iso_equiv_nonzero : ∀ {Y : C} (f : Y ⟶ X) [mono f], is_iso.{v} f ≃ (f ≠ 0))
+
+/-- A nonzero monomorphism to a simple object is an isomorphism. -/
+def is_iso_of_mono_of_nonzero {X Y : C} [simple.{v} X] {f : Y ⟶ X} [mono f] (w : f ≠ 0) :
+  is_iso f :=
+(simple.mono_is_iso_equiv_nonzero f).symm w
+
+lemma kernel_zero_of_nonzero_from_simple
+  {X Y : C} [simple.{v} X] {f : X ⟶ Y} [has_limit (parallel_pair f 0)] (w : f ≠ 0) :
+  kernel.ι f = 0 :=
+begin
+  classical,
+  by_contradiction h,
+  haveI := is_iso_of_mono_of_nonzero h,
+  exact w (eq_zero_of_kernel_is_iso f),
+end
+
```

Comment:
Could you add the dual lemmas and constructor? Every nonzero epi out of a simple object is an iso. Or is this only true in abelian categories?

Reply:
This, AFAIK, is only true in an abelian category, but I've now added it.